streaming_form_data/_parser.pyx

```diff
@@ -2,42 +2,59 @@ import cgi
 
 from streaming_form_data.targets import NullTarget
 
+ctypedef unsigned char Byte
+ctypedef size_t Index
 
 cdef enum Constants:
     Hyphen = 45
     CR = 13
     LF = 10
-    MaxBufferSize = 1024
+    MinFileBodyChunkSize = 1024
 
 
 cdef enum FinderState:
     FS_START, FS_WORKING, FS_END
 
 
+# Knuth-Morris-Pratt algorithm
 cdef class Finder:
     cdef bytes target
-    cdef long index
+    cdef const Byte *target_ptr
+    cdef Index target_len, index
     cdef FinderState state
 
     def __init__(self, target):
         if len(target) < 1:
             raise ValueError('Empty values not allowed')
 
         self.target = target
+        self.target_ptr = self.target
+        self.target_len = len(self.target)
         self.index = 0
         self.state = FinderState.FS_START
 
-    cpdef feed(self, long byte):
-        if byte != self.target[self.index]:
-            self.state = FinderState.FS_START
-            self.index = 0
+    cpdef feed(self, Byte byte):
+        if byte != self.target_ptr[self.index]:
+            if self.state != FinderState.FS_START:
+                self.state = FinderState.FS_START
+                self.index = 0
+                # try matching substring
+                # This is not universal code, but the code specialized from multipart delimiters
+                # (length is at least 5 bytes, starting with \r\n and has no \r\n in the middle)
+                if byte == self.target_ptr[0]:
+                    self.state = FinderState.FS_WORKING
+                    self.index = 1
         else:
             self.state = FinderState.FS_WORKING
             self.index += 1
 
-            if self.index == len(self.target):
+            if self.index == self.target_len:
                 self.state = FinderState.FS_END
 
+    cpdef force_match_4chars(self):
```

Comment:
Which 4 characters are these? CRLF?

Reply:
"\r\n--"
We can pass 4 as 'count' argument instead.

Reply:
hmm. I can also get rid of this method. And use reset() + feed() x 4

Reply:
`reset() + feed() * 4` sounds good. Please also add a comment where you do this - this could be tricky to catch.